src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs

```diff
@@ -269,7 +270,12 @@ private async Task<Operation> ImportInternalAsync(DatabaseSmugglerImportOptions
             }
             catch (Exception e)
             {
-                returnContext?.Dispose();
+                if (requestTask == null)
+                {
+                    // handle the possible double dispose of return context
+                    returnContext?.Dispose();
```

Comment:
Double dispose should always be safe, no?

Reply:
double dispose of the context is safe, but not the dispose of the `ReturnRequestContext`, even if the dispose is not concurrent (it isn't thread safe) you might hit different dispose path if the low mem event going on and off

Reply:
There should be no case that the dispose here is concurrent. 
And the `ReturnRequestContext` should be made for repeated calls.